src/App.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import { BrowserRouter as Router, Route, Switch} from 'react-router-dom'
+import authentication from 'react-azure-adb2c'
+
+import { 
+  LoginPage as Homepage, 
+  StudentsPage, 
+  InstProfilePage, 
+  InstOfferingPage,
+  VideoPage,
+  AdminPage,
+} from './pages'
+import './App.css'
+import 'semantic-ui-css/semantic.min.css'
+
+
+class App extends React.Component {
+
+  render() {
+    const NotFound404 = <div>404</div>
+    return (
+      <Router basename="/">
+        <Switch>
+          <Route exact path="/" component={Homepage} />
+          <Route path="/admin" component={authentication.required(AdminPage)} />
+          <Route path="/student" component={authentication.required(StudentsPage)} />
```

Comment:
We have a flow for Public Users,
Where a user if he has not authenticated, he can still view all public courses
So you could remove the authentication required requirement from this.
The courses you'll display are all which are publicly accessible.

If this isn't clear or it'll require quite a bit of rewiring, talk to me later about this

Reply:
So for the public courses, will they also be open to people who don't log in?